src/plugins/telemetry/server/telemetry_collection/get_data_telemetry/get_data_telemetry.ts

```diff
@@ -171,95 +131,85 @@ export function buildDataTelemetryPayload(indices: DataTelemetryIndex[]): DataTe
+            '*.mappings._meta.managed_by',
+            // Does it have `ecs.version` in the mappings? => It follows the ECS conventions
+            '*.mappings.properties.ecs.properties.version.type',
+
+            // If `data_stream.type` is a `constant_keyword`, it can be reported as a type
+            '*.mappings.properties.data_stream.properties.type.value',
+            // If `data_stream.dataset` is a `constant_keyword`, it can be reported as the dataset
+            '*.mappings.properties.data_stream.properties.dataset.value',
+          ],
+        }),
+        // GET <index>/_stats/docs,store?level=indices&filter_path=indices.*.total
+        callCluster<IndexStats>('indices.stats', {
+          index,
+          level: 'indices',
+          metric: ['docs', 'store'],
+          filterPath: ['indices.*.total'],
+        }),
+      ]);
+      const indexNames = Object.keys({ ...indexMappings, ...indexStats?.indices });
+    }
 
-          // If `data_stream.type` is a `constant_keyword`, it can be reported as a type
-          '*.mappings.properties.data_stream.properties.type.value',
-          // If `data_stream.dataset` is a `constant_keyword`, it can be reported as the dataset
-          '*.mappings.properties.data_stream.properties.dataset.value',
-        ],
-      }),
-      // GET <index>/_stats/docs,store?level=indices&filter_path=indices.*.total
-      callCluster<IndexStats>('indices.stats', {
-        index,
-        level: 'indices',
-        metric: ['docs', 'store'],
-        filterPath: ['indices.*.total'],
-      }),
+    const indexMappingsParams: RequestParams.IndicesGetMapping = {
+      index: '*',
+      filter_path: [
+        // _meta.beat tells the shipper
+        '*.mappings._meta.beat',
+        // _meta.package.name tells the Ingest Manager's package
+        '*.mappings._meta.package.name',
+        // _meta.managed_by is usually populated by Ingest Manager for the UI to identify it
+        '*.mappings._meta.managed_by',
+        // Does it have `ecs.version` in the mappings? => It follows the ECS conventions
+        '*.mappings.properties.ecs.properties.version.type',
+
+        // If `data_stream.type` is a `constant_keyword`, it can be reported as a type
+        '*.mappings.properties.data_stream.properties.type.value',
+        // If `data_stream.dataset` is a `constant_keyword`, it can be reported as the dataset
+        '*.mappings.properties.data_stream.properties.dataset.value',
+      ],
+    };
+    const indicesStatsParams: RequestParams.IndicesStats = {
+      index: 'index',
+      level: 'indices',
+      metric: ['docs', 'store'],
+      filter_path: ['indices.*.total'],
+    };
+    const [indexMappingsResponse, indexStatsResponse]: [
+      ApiResponse,
+      ApiResponse
+    ] = await Promise.all([
+      esClient.indices.getMapping(indexMappingsParams),
+      esClient.indices.stats(indicesStatsParams),
```

Comment:
```suggestion
    const [{ body: indexMappings }, { body: indexStats }] = await Promise.all([
      esClient.indices.getMapping<IndexMappings>(indexMappingsParams),
      esClient.indices.stats<IndexStats>(indicesStatsParams),
```

So we don't need the castings in lines 208-209 🙂